core/arch/arm/kernel/thread.c

```diff
@@ -1647,7 +1647,7 @@ static struct mobj *get_rpc_alloc_res(struct optee_msg_arg *arg,
 
 	psize = READ_ONCE(arg->params[0].u.tmem.size);
 	if (psize < size)
-		return NULL;
+		goto err;
```

Comment:
We need a cookie for the call to `thread_rpc_free()` below to make sense.
This is a breach of contract so it should never happen, I'm not sure how effective this can be.

Reply:
Oh, right :/ 

Will fix this.

Actually, I want this patch, because it will ease up op-tee mediator in Xen. There is a case, when guest provides a buffer to OP-TEE, but Xen can't do translation for some reason (lack of resources for example). Xen could report error to OP-TEE, but it is hard to issue `OPTEE_RPC_CMD_SHM_FREE` to guest from Xen side. So, I thought than Xen can present zero-length buffer to OP-TEE and OP-TEE will call `thread_rpc_free()`.

Reply:
Fair enough. Please leave a comment explaining this.